main.go

```diff
@@ -38,45 +39,79 @@ var (
 	signingNameOverride = kingpin.Flag("name", "AWS Service to sign for").String();
 	hostOverride = kingpin.Flag("host", "Host to proxy to").String();
 	regionOverride = kingpin.Flag("region", "AWS region to sign for").String();
+	configSets = kingpin.Flag("config-set", "Host-based configuration overrides for role-arn/name/host/region (encoded as yaml)").Short('c').Strings()
 )
 
-func main() {
-	kingpin.Parse()
-
-	log.SetLevel(log.InfoLevel)
-	if *debug {
-		log.SetLevel(log.DebugLevel)
-	}
 
+func getSigner(roleArn string) *v4.Signer {
     session, err := session.NewSession()
     if err != nil {
         log.Fatal(err)
     }
 
 	var credentials *credentials.Credentials
-	if *roleArn != "" {
-	    credentials = stscreds.NewCredentials(session, *roleArn, func(p *stscreds.AssumeRoleProvider) {
+	if roleArn != "" {
+	    credentials = stscreds.NewCredentials(session, roleArn, func(p *stscreds.AssumeRoleProvider) {
 		    p.RoleSessionName = roleSessionName()
 	    })
 	} else {
 	    credentials = session.Config.Credentials
 	}
 
-	signer := v4.NewSigner(credentials)
+	return v4.NewSigner(credentials)
+}
+
+func main() {
+	kingpin.Parse()
+
+	log.SetLevel(log.InfoLevel)
+	if *debug {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	signer := getSigner(*roleArn)
 
 	log.WithFields(log.Fields{"StripHeaders": *strip}).Infof("Stripping headers %s", *strip)
 	log.WithFields(log.Fields{"port": *port}).Infof("Listening on %s", *port)
 
+	clients := map[string]handler.Client{
+		"default": &handler.ProxyClient{
```

Comment:
This is not ideal, the handler is always expecting the default to be set, and it's manually set in the main func. Ideally, you'd use a separate constructor parameter, or (since the whole aws implementation is not that well structured) just add an error check in the handler when getting the `default` val so it breaks if anyone changes the main function call

Reply:
Added the runtime check